client/allocrunner/taskrunner/template/template.go

```diff
@@ -509,7 +509,7 @@ func templateRunner(config *TaskTemplateManagerConfig) (
 	}
 
 	// Set Nomad's environment variables
-	runner.Env = config.EnvBuilder.Build().All()
+	runner.Env = maskProcessEnv(config.EnvBuilder.Build().All())
```

Comment:
Need a comment explaining why we need to mask.

Reply:
I'm going to carry this PR because @notnoop is on vacation. Pushed that commit. Looks like the test failure in https://travis-ci.org/hashicorp/nomad/jobs/566415765 might be a flaky test? I'll follow up on that later as separate work.